packages/Uploader/src/components/File/File.js

```diff
@@ -23,45 +24,46 @@ const propTypes = {
   error: PropTypes.string,
   fileKey: PropTypes.string.isRequired,
   name: PropTypes.string.isRequired,
+  onError: PropTypes.func,
+  onCancel: PropTypes.func,
   progress: PropTypes.number,
   size: PropTypes.number.isRequired,
   status: PropTypes.oneOf(Object.keys(File.types.status).map(key => File.types.status[key])).isRequired, // eslint-disable-line no-use-before-define
 };
 
 const defaultProps = {
   error: "",
+  onError: null,
+  onCancel: null,
   progress: 0,
 };
 
-function File({ error, fileKey, name, progress, size, status }) {
-  const { cancelFile, restartFileUpload } = React.useContext(UploaderContext);
+function File(props) {
+  const { error, fileKey, name, progress, size, status, onError, onCancel } = props;
+
+  const { cancelFile } = React.useContext(UploaderContext);
   const I18n = useI18n();
   const sizeWithUnits = getNumberWithUnits(I18n, size);
   const progressWithUnits = getNumberWithUnits(I18n, (size * progress) / 100);
 
   function renderIcon() {
     switch (status) {
       case types.status.ERROR:
-      case types.status.CANCEL:
         return (
           <Popover isDark isEager>
             <Popover.Tip />
             <Popover.Trigger>
-              <Button.Icon
-                kind={Button.Icon.types.kind.MINOR}
-                onClick={() => {
-                  restartFileUpload(fileKey);
-                }}
-                size={Button.Icon.types.size.SMALL}
-              >
-                <RetryIcon />
+              <Button.Icon kind={Button.Icon.types.kind.MINOR} onClick={() => {}}>
```

Comment:
Will this render two nested buttons? one for trigger, one for button.icon, I usually use `<Popover.Trigger>{() => <Button />}</Popover.Trigger>` for a case like this. But I can see this is not an issue introduced by this PR, so I guess it's ok to ignore it?

Reply:
Yeah I think might render two buttons didn't notice before, I still have to work on this component after some decisions, will fix it in the next PR.